plumbing/format/packfile/parser.go

```diff
@@ -271,6 +300,12 @@ func (p *Parser) resolveDeltas() error {
 		if err != nil {
 			return err
 		}
+
+		if obj.DiskType.IsDelta() {
+			p.deltasSeen++
+			p.writeDeltaProgress()
+		}
```

Comment:
🍜 The key here was that `obj.DiskType` will indicate that it's a delta, rather than using `obj.Type` (for reasons I don't really understand).